guides/common/modules/proc_securing-the-dhcp-api.adoc

```diff
@@ -0,0 +1,31 @@
+[id="Securing_the_dhcpd_API_{context}"]
+= Securing the dhcpd API
+
+By default, the dhcpd API listens to any host.
+You can add a `omapi_key` to provide basic security.
+
+.Procedure
+. Generate a key:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+# yum install -y bind
+# dnssec-keygen -r /dev/urandom -a HMAC-MD5 -b 512 -n HOST omapi_key
+# cat Komapi_key.+*.private | grep ^Key|cut -d ' ' -f2-
+----
+. Edit your `/etc/dhcpd.conf`:
++
+[options="nowrap", subs="+quotes,verbatim,attributes"]
+----
+omapi-port 7911;
+key omapi_key {
+algorithm HMAC-MD5;
+  secret "XXXXXXXXX"; #<-The output from the generated key above.
+};
+omapi-key omapi_key;
+----
+. Ensure to add the omapi_key to `dhcp_isc.yml` on your {SmartProxyServer}.
+. Restart the `dhcpd` and `foreman-proxy` services.
+. In the {ProjectWebUI}, navigate to *Infrastructure > Smart Proxies*.
+. Select your {SmartProxy}.
+. On the *Overview* tab, click *Refresh* in the *Actions* menu.
```

Comment:
The proper way to do this is using the installer:
```
foreman-installer \
  --foreman-proxy-dhcp-key-name "myname" \
  --foreman-proxy-dhcp-key-secret "mysecret"
```

We don't do this by default (because generating a secret and parsing that was hard) but surprisingly enough we don't document it either.

Reply:
Applied!